From 77c960bebb34dc661ce2d1de3cac983b7c1b3f61 Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Tue, 27 May 2025 10:02:12 +0300
Subject: [PATCH] [SW-224445] g3: lfs: use time critical log mode

Use time critical logging mode for FW agent.
If is important, because communication protocol
between host and agent is time critical, in this
mode heavy operation (attempt to repair file
system, log file swapping) will not be executed.

Change-Id: I50c77be15025e6cf692121b29eb5344cdb41ef36
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/fw-loader/src/fw_upd.c b/fw-loader/src/fw_upd.c
index 29dce17..a0b3366 100644
--- a/fw-loader/src/fw_upd.c
+++ b/fw-loader/src/fw_upd.c
@@ -58,6 +58,10 @@
 #include <pldm_fu.h>
 #endif
 
+#ifdef LFS_LOG_ENABLED
+#include "lfs_zephyr_init.h"
+#endif
+
 #define MODULE_FWUPD_STACKSIZE	(8192 - CONFIG_HL_STACK_OVERFLOW_REDUCE_SIZE)
 #define FWUPD_WAIT_THREAD_STOP_MSEC	K_MSEC(1000)
 static k_tid_t fwupd_thread_id;
@@ -710,6 +714,15 @@
 	return CMD_RET_FAILURE;
 }
 
+#ifdef LFS_LOG_ENABLED
+static void shutdown_lfs_log(void)
+{
+	module_lfs_shutdown();
+}
+#else
+static void shutdown_lfs_log(void) {}
+#endif
+
 static void fwupd_reset_device(void)
 {
 	fwupd_send_status(HL_UPD_STS_RST_ACK, 0);
@@ -722,6 +735,7 @@
 		hl_force_recovery_flow();
 	}
 
+	shutdown_lfs_log();
 	hl_i2c_master_disable();
 	hl_i2c_slave_disable();
 	hl_dev_reset();
diff --git a/fw-loader/src/main.c b/fw-loader/src/main.c
index 85cfd58..ea45152 100644
--- a/fw-loader/src/main.c
+++ b/fw-loader/src/main.c
@@ -206,7 +206,7 @@
 #ifdef CONFIG_LFS_ENABLED
 static void start_lfs_logger(void)
 {
-	lfs_console_init();
+	lfs_console_init_time_critical();
 }
 #else
 static void start_lfs_logger(void) {}
